bftengine/src/bftengine/SigManager.cpp

```diff
@@ -215,7 +215,12 @@ bool SigManager::verifySig(
     idOfReplica = replicasInfo_.isIdOfReplica(pid);
   }
   idOfReadOnlyReplica = replicasInfo_.isIdOfPeerRoReplica(pid);
-  ConcordAssert(idOfReplica || idOfExternalClient || idOfReadOnlyReplica);
+  if (!(idOfReplica || idOfExternalClient || idOfReadOnlyReplica)) {
+    std::stringstream msg;
+    msg << "Invalid ID of replicas : " << KVLOG(idOfReplica, idOfExternalClient, idOfReadOnlyReplica);
+    LOG_ERROR(GL, msg.str());
```

Comment:
why do we need the std::stringstream?
`LOG_ERROR(GL, "Invalid ID of replicas:" << KVLOG(idOfReplica, idOfExternalClient, idOfReadOnlyReplica));`
please pay attention: KVLOG already adds a space after the ':'.